crates/oxc_minifier/src/compressor/fold.rs

```diff
@@ -422,4 +431,29 @@ impl<'a> Compressor<'a> {
         }
         None
     }
+
+    fn try_fold_and_or<'b>(
+        &mut self,
+        _span: Span,
+        op: LogicalOperator,
+        left_expr: &'b Expression<'a>,
+        _right_expr: &'b Expression<'a>,
+    ) -> Option<Expression<'a>> {
+        let boolean_value = get_boolean_value(left_expr);
+        if let Some(boolean_value) = boolean_value {
+            // (TRUE || x) => TRUE (also, (3 || x) => 3)
+            // (FALSE && x) => FALSE
+            if (boolean_value && op == LogicalOperator::Or)
+                || (!boolean_value && op == LogicalOperator::And)
+            {
+
+                // without knowing left_expr/right_expr's kind, how can I return it?
+                // return Some(left_expr)
```

Comment:
@Boshen Hey Boshen, do you have advice about how to return the child of expression without recreating it?

Related code in [ closure-compiler](https://github.com/google/closure-compiler/blob/e13f5cd0a5d3d35f2db1e6c03fdf67ef02946009/src/com/google/javascript/jscomp/PeepholeFoldConstants.java#L602-L604).

Reply:
I think you need to own the logical expression here? i.e. pass in logical expression by value.